app/backend/approaches/retrievethenreadvision.py

```diff
@@ -118,46 +110,32 @@ async def run(
             minimum_reranker_score,
         )
 
-        image_list: list[ChatCompletionContentPartImageParam] = []
-        user_content: list[ChatCompletionContentPartParam] = [{"text": q, "type": "text"}]
-
         # Process results
-        sources_content = self.get_sources_content(results, use_semantic_captions, use_image_citation=True)
-
+        text_sources = []
+        image_sources = []
         if send_text_to_gptvision:
-            content = "\n".join(sources_content)
-            user_content.append({"text": content, "type": "text"})
+            text_sources = self.get_sources_content(results, use_semantic_captions, use_image_citation=True)
         if send_images_to_gptvision:
             for result in results:
                 url = await fetch_image(self.blob_container_client, result)
                 if url:
-                    image_list.append({"image_url": url, "type": "image_url"})
-            user_content.extend(image_list)
+                    image_sources.append(url)
 
-        response_token_limit = 1024
-        updated_messages = build_messages(
```

Comment:
Same comment here, I removed the build_messages call since it isn't useful when there's no history being passed in.